cli/pkg/kctrl/cmd/app/init/configure/fetch/vendir.go

```
@@ -0,0 +1,218 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package fetch
+
+import (
+	"bytes"
+	"fmt"
+	goexec "os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/app/init/common"
+	cmdcore "github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/core"
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/exec"
+	vendirconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
+)
+
+const (
+	VendirFileName      = "vendir.yml"
+	VendirSyncDirectory = "upstream"
+	IncludeAllFiles     = "*"
+)
+
+type VendirStep struct {
+	ui          cmdcore.AuthoringUI
+	config      vendirconf.Config
+	fetchOption string
+}
+
+func NewVendirStep(ui cmdcore.AuthoringUI, config vendirconf.Config, fetchOption string) *VendirStep {
+	vendirStep := VendirStep{
+		ui:          ui,
+		config:      config,
+		fetchOption: fetchOption,
+	}
+	return &vendirStep
+}
+
+func (v *VendirStep) PreInteract() error {
+	return nil
+}
+
+func (v *VendirStep) Interact() error {
+	vendirDirectories := v.config.Directories
+	if len(vendirDirectories) > 1 {
+		//TODO what if we have >1 Directories section in the vendir conf
```

If the vendir.yml has more than 1 directories, then should we return from here as this is a advance use case for the tool? This mean we wont ask for question, but we will still run `vendir sync`

---

lets just return an error. i dont think we should run vendir sync (at least at this point).

---

Adopted.